tdrs-backend/tdpservice/data_files/views.py

```diff
@@ -122,8 +139,21 @@ def download(self, request, pk=None):
         """Retrieve a file from s3 then stream it to the client."""
         record = self.get_object()
 
+        # If no versioning id, then download from django storage
+        if not hasattr(record, 's3_versioning_id') or record.s3_versioning_id is None:
+            response = FileResponse(
+                FileWrapper(record.file),
+                filename=record.original_filename
+            )
+            return response
+
+        # If versioning id, then download from s3
+        s3 = S3Client()
+        file_path = record.file.name
+        version_id = record.s3_versioning_id
+
         response = FileResponse(
-            FileWrapper(record.file),
+            FileWrapper(s3.file_download(file_path, record.original_filename, version_id)),
             filename=record.original_filename
         )
```

Comment:
Would removing the first `return response` and putting this second-half in an `else` block be more readable to you? Leaves us with the single return regardless of which method is used